plugins/catalog-backend/src/ingestion/CatalogRules.ts

```diff
@@ -92,9 +93,23 @@ export class DefaultCatalogRulesEnforcer implements CatalogRulesEnforcer {
     const rules = new Array<CatalogRule>();
 
     if (config.has('catalog.rules')) {
-      const globalRules = config.getConfigArray('catalog.rules').map(sub => ({
-        allow: sub.getStringArray('allow').map(kind => ({ kind })),
-      }));
+      const globalRules = config
+        .getConfigArray('catalog.rules')
+        .map(ruleConfig => {
+          const rule: CatalogRule = {
+            allow: ruleConfig.getStringArray('allow').map(kind => ({ kind })),
+          };
+
+          const locConf = ruleConfig.getOptionalConfigArray('locations');
```

Comment:
You could actually chain all of this up with `locations: ruleConfig.getOptionalConfigArray('locations')?.map(...)`, thinking that would be cleaner.